SalesforceFoundation/EDA
# [EDA] Prevent Deletion of a Behavior Involvement Record when there is a Related Behavior Response Record

# Critical Changes

# Changes
We've added functionality to prevent the deletion of Behavior Involvement records that have related Behavior Response records. This change is for internal-use only and is disabled at this time. You won't see any difference in functionality.
**Note:** We strongly recommend that EDA Admins don't manage Hierarchy Custom Settings directly and, instead, use the EDA Settings UI.

# Issues Closed
Closes #1102

# New Metadata

# Deleted Metadata
BEH_CannotDelete_TDTM
BEH_CannotDelete_TEST

# Testing Notes
Scenario 1:  “Prevent Behavior Involvement Deletion” checkbox is enabled and Behavior Involvement has related Behavior Response child record.
- Check “Prevent Behavior Involvement Deletion” checkbox in Hierarchy Settings.
- Create multiple Contact records  
- Create multiple Case records and associate the Contacts to the new Case records 
- Create multiple Behavior Involvement records and associate the Contacts and Cases created previously
- Create multiple Behavior Responses and associate to the Behavior Involvements 
- Delete Behavior Involvement records
- Result: You should receive a pop up error that you cannot delete this Behavior Involvement records

Scenario 2: “Prevent Behavior Involvement Deletion” checkbox is enabled and Behavior Involvement has NO related Behavior Response child record.
- Check “Prevent Behavior Involvement Deletion” checkbox in Hierarchy Settings.
- Create multiple Contact records  
- Create multiple Case records and associate the Contacts to the new Case records 
- Create multiple Behavior Involvement records and associate the Contacts and Cases created previously
- Delete Behavior Involvement records
- Result: You not receive a pop up error and should be able to delete the Behavior Involvement records. 

Scenario 3:  “Prevent Behavior Involvement Deletion” checkbox is disabled and Behavior Involvement has related Behavior Response child record.
- Check “Prevent Behavior Involvement Deletion” checkbox in Hierarchy Settings.
- Create multiple Contact records  
- Create multiple Case records and associate the Contacts to the new Case records 
- Create multiple Behavior Involvement records and associate the Contacts and Cases created previously
- Create multiple Behavior Responses and associate to the Behavior Involvements 
- Delete Behavior Involvement records
- Result: You not receive a pop up error and should be able to delete the Behavior Involvement records. 

Scenario 4: “Prevent Behavior Involvement Deletion” checkbox is disabled and Behavior Involvement has NO related Behavior Response child record.
- Check “Prevent Behavior Involvement Deletion” checkbox in Hierarchy Settings.
- Create multiple Contact records  
- Create multiple Case records and associate the Contacts to the new Case records 
- Delete Behavior Involvement records
- Result: You not receive a pop up error and should be able to delete the Behavior Involvement records.
